### ISSUE TITLE
Support for embedded expressions in plaintext templates

### ISSUE THREAD

**Issue by beerlington** [maizzle/framework#394]
- Maizzle Version: 3.1.0
- Node.js Version: 12.14.1

Is there a way to get embedded expressions (e.g. Rails or Phoenix templates) to work when compiling plaintext templates?

If I have something like this in my source template:

```eex
<div>
  My variable: <%= @variable_name %>
</div>
```

I would expect it to produce a plaintext version that removes just the `div` tag like this:

```
My variable: <%= @variable_name %>
```

but instead it removes the embedded expression too and is left with:

```
My variable:
```

I experimented a little with the [`cb` option](https://codsen.com/os/string-strip-html/#optional-options-object) in string-strip-html by ignoring what it thinks are `%` tag names. It works in simple cases, but breaks down when the embedded expression is an attribute in an `a` or `img` tag such as the link href or img src.

Here's an example of what I tried for the plaintext config:

```js
  plaintext: {
    destination: {
      extension: 'text.eex',
    },
    cb: ({
      tag,
      deleteFrom,
      deleteTo,
      insert,
      rangesArr,
      proposedReturn,
    }) => {
      if (tag.name !== '%') {
        rangesArr.push(deleteFrom, deleteTo, insert);
      }
    }
  }
```

That works with something simple like my `div` example above, but produces strange output for tags where an attribute is dynamic, like these:

```eex
<img src="<%= @image_url %>" style="border:0" alt="">
```
or
```eex
<a href="<%= @support_url %>" style="color:#FFFFFF;"><span style="color:#FFFFFF!">Have Questions?</span></a>
```

Any thoughts on how we might be able to handle the more complex cases? They work perfectly when compiling HTML templates.

**Comment by cossssmin** [comment]
Yeah, looks like `string-strip-html` is tripping on those tags. Pinged the author, let's wait to see what they say...

**Comment by beerlington** [comment]
@cossssmin I was curious if you heard back on this.

**Comment by cossssmin** [comment]
Not yet @beerlington, but the author is now back on GitHub so I've opened the issue which you can see above.

**Comment by cossssmin** [comment]
@beerlington fixed in `string-strip-html@8.1.0`, will be part of the next Maizzle release.

**Comment by cossssmin** [comment]
Fixed in [v3.1.4](https://github.com/maizzle/framework/releases/tag/v3.1.4)

**Comment by beerlington** [comment]
@cossssmin this fixes most of the cases, but we just hit a weird edge case with links that have more than one dynamic parameter.

This version works:

```eex
<a href="https://example.com/test?param1=<%= @param1 %>">click me</a>
```

which correctly produces:

```
click me
https://example.com/test?param1=<%= @param1 %>
```

However, when there are multiple params in the URL, it produces something funky:

```ex
<a href="https://example.com/test?param1=<%= @param1 %>&param2=<%= @param2 %>">click me</a>
```

produces:
```
¶m2=<%= @param2 %>">click me
```

**Comment by cossssmin** [comment]
Can confirm, I've let the author know in https://github.com/codsen/codsen/issues/2

**Comment by cossssmin** [comment]
@beerlington fixed in `string-strip-html@8.2.0`, added in Maizzle [v3.1.5](https://github.com/maizzle/framework/releases/tag/v3.1.5)
